lib/src/screens/dashboard/desktop_widgets/desktop_sidebar_wrapper.dart

```diff
@@ -77,27 +77,25 @@ class DesktopSidebarWrapper extends BasePage {
               SideMenuItem(
                 imagePath: 'assets/images/wallet_outline.png',
                 isSelected: desktopSidebarViewModel.currentPage == SidebarItem.dashboard,
-                onTap: () => desktopSidebarViewModel.onPageChange(SidebarItem.dashboard),
+                onTap: () {
+                  desktopSidebarViewModel.onPageChange(SidebarItem.dashboard);
+                  desktopNavigatorKey.currentState
+                      ?.pushNamedAndRemoveUntil(Routes.desktop_actions, (route) => false);
+                },
               ),
               SideMenuItem(
                 onTap: () {
-                  String? currentPath;
-
-                  desktopNavigatorKey.currentState?.popUntil((route) {
-                    currentPath = route.settings.name;
-                    return true;
-                  });
-
-                  switch (currentPath) {
-                    case Routes.transactionsPage:
-                      desktopSidebarViewModel.resetSidebar();
-                      break;
-                    default:
-                      desktopSidebarViewModel.resetSidebar();
-                      Future.delayed(Duration(milliseconds: 10), () {
-                        desktopSidebarViewModel.onPageChange(SidebarItem.transactions);
-                        desktopNavigatorKey.currentState?.pushNamed(Routes.transactionsPage);
-                      });
+                  if (desktopSidebarViewModel.currentPage == SidebarItem.transactions) {
+                    desktopNavigatorKey.currentState
+                        ?.pushNamedAndRemoveUntil(Routes.desktop_actions, (route) => false);
+                    desktopSidebarViewModel.resetSidebar();
+                  } else {
+                    desktopSidebarViewModel.resetSidebar();
+                    Future.delayed(Duration(milliseconds: 10), () {
+                      desktopSidebarViewModel.onPageChange(SidebarItem.transactions);
+                      desktopNavigatorKey.currentState
+                          ?.pushNamedAndRemoveUntil(Routes.transactionsPage, (route) => false);
```

Comment:
I believe there is no need to resetSidebar and this is the cause for the error, so can you try and remove this line `desktopSidebarViewModel.resetSidebar();` and remove the `Future.delayed` and try, and let me know how it goes